bluesky_widgets/qt/search_input.py

```diff
@@ -16,54 +26,138 @@ def __init__(self, model, *args, **kwargs):
+        self.radio_button_group = QButtonGroup()
+        self.radio_button_group.addButton(self.all_widget)
+        self.radio_button_group.addButton(self.days_widget)
+        self.radio_button_group.addButton(self.today_widget)
+        self.radio_button_group.addButton(self.hour_widget)
+        default_period_layout = QGridLayout()
+        default_period_layout.addWidget(self.all_widget, 0, 0, 1, 1)
+        default_period_layout.addWidget(self.days_widget, 0, 1, 1, 1)
+        default_period_layout.addWidget(self.today_widget, 1, 0, 1, 1)
+        default_period_layout.addWidget(self.hour_widget, 1, 1, 1, 1)
+        self.layout().addRow("When:", default_period_layout)
 
         # "Since: <datetime picker>"
         self.since_widget = QDateTimeEdit()
         self.since_widget.setCalendarPopup(True)
         self.since_widget.setDisplayFormat("yyyy-MM-dd HH:mm")
         self.layout().addRow("Since:", self.since_widget)
+        self.since_widget.dateTimeChanged.connect(self.uncheck_radiobuttons)
 
         # "Until: <datetime picker>"
         self.until_widget = QDateTimeEdit()
         self.until_widget.setCalendarPopup(True)
         self.until_widget.setDisplayFormat("yyyy-MM-dd HH:mm")
         self.layout().addRow("Until:", self.until_widget)
+        self.until_widget.dateTimeChanged.connect(self.uncheck_radiobuttons)
 
+        # Refresh Button
         self.refresh_button = QPushButton("Refresh")
         self.layout().addWidget(self.refresh_button)
 
         # Initialize values.
         qdatetime = QDateTime()
-        qdatetime.setSecsSinceEpoch(self.model.since)
+        qdatetime.setSecsSinceEpoch(self.model.since.timestamp())
         self.since_widget.setDateTime(qdatetime)
         qdatetime = QDateTime()
-        qdatetime.setSecsSinceEpoch(self.model.until)
+        qdatetime.setSecsSinceEpoch(self.model.until.timestamp())
         self.until_widget.setDateTime(qdatetime)
 
         # Changes to the GUI update the model.
         self.since_widget.dateTimeChanged.connect(self.on_since_view_changed)
         self.until_widget.dateTimeChanged.connect(self.on_until_view_changed)
-        self.refresh_button.clicked.connect(self.model.events.reload)
+        # TODO: check what refresh button is doing
+        # self.refresh_button.clicked.connect(self.model.events.reload)
+        self.refresh_button.clicked.connect(self.on_refresh_clicked)
         # Changes to the model update the GUI.
         self.model.events.since.connect(self.on_since_model_changed)
         self.model.events.until.connect(self.on_until_model_changed)
 
+        # connect QRadioButtons and change date dropdowns (since/until widgets) accordingly
+        self.today_widget.clicked.connect(self.on_select_today)
+        self.hour_widget.clicked.connect(self.on_select_lasthour)
+        self.days_widget.clicked.connect(self.on_select_30days)
+        self.all_widget.clicked.connect(self.on_select_all)
```

Comment:
Here in `__init__` we should configure `model` to only accept certain timedelta values (the ones which the particular UI can display) and reject any others with a `ValueError`. I sketched the code to do so in a test: I think all that's left to do is copy/paste [this test code](https://github.com/bluesky/bluesky-widgets/pull/13/files#diff-b58e3b3ad1d5c95fdfb98a50d43a9972R88-R113) here in `__init__`.

Reply:
Ok, this restriction is only needed for input from the CL. Is the simple copy/paste sufficient for your purpose?

Reply:
The 'all' timedelta does not fit too well here, since all is an arbitrary definition we (or I atm made, did you figure out what date I chose? ;-))

Reply:
Hmm...is that 1866?

Reply:
1815-12-10... A birthday

Reply:
Oh I see that's relative to 1970 of course, not relative to today. Ada!